sunshine/confighttp.cpp

```diff
@@ -87,6 +96,13 @@ bool authenticate(resp_https_t response, req_https_t request) {
     send_unauthorized(response, request);
   });
 
+  //If credentials are shown, redirect the user to a /welcome page
+  if(config::sunshine.showCredentials){
```

Comment:
Instead of using showCredentials.
You could: `if(config::sunshine.username.empty())`
And modifying existing code so that it no longer automatically creates credentials.
Automatically creating credentials is obsolete with this new method of creating a password.

Also, it may be worth setting this piece if code before:
```
auto fg = util::fail_guard([&]() {
  send_unauthorized(response, request);
});
```
No need for `fg.disable()`

Reply:
I like this idea, I'll change the code to allow the user to set the credentials directly